Sources/TuistKit/Services/Inspect/InspectBuildService.swift

```diff
@@ -0,0 +1,197 @@
+        }
+        let xcactivityLog = try await latestXCActivityLog(
+            logManifestPlistPath: logManifestPlistPath,
+            buildLogsPath: buildLogsPath,
+            projectPath: projectPath,
+            referenceDate: referenceDate
+        )
+        try await createBuild(
+            for: xcactivityLog,
+            projectPath: projectPath
+        )
+    }
+
+    private func createBuild(
+        for xcactivityLog: XCActivityLog,
+        projectPath: AbsolutePath
+    ) async throws {
+        let config = try await configLoader
+            .loadConfig(path: projectPath)
+        guard let fullHandle = config.fullHandle else { throw InspectBuildServiceError.missingFullHandle }
+        try await createBuildService.createBuild(
+            fullHandle: fullHandle,
+            serverURL: config.url,
+            id: xcactivityLog.mainSection.uniqueIdentifier,
+            duration: Int(xcactivityLog.mainSection.timeStoppedRecording * 1000) -
+                Int(xcactivityLog.mainSection.timeStartedRecording * 1000),
+            isCI: ciChecker.isCI(),
+            modelIdentifier: machineEnvironment.modelIdentifier(),
+            macOSVersion: machineEnvironment.macOSVersion,
+            scheme: environment.schemeName,
+            xcodeVersion: try await xcodeBuildController.version()?.description
+        )
+        ServiceContext.current?.ui?.success(
+            .alert(
+                "Uploaded a build to the server."
+            )
+        )
+    }
+
+    private func latestXCActivityLog(
+        logManifestPlistPath: AbsolutePath,
+        buildLogsPath: AbsolutePath,
+        projectPath: AbsolutePath,
+        referenceDate: Date
+    ) async throws -> XCActivityLog {
+        let plist: LogStoreManifest = try await fileSystem
+            .readPlistFile(at: logManifestPlistPath)
+
+        guard let latestLog = plist.logs.values.sorted(by: { $0.timeStoppedRecording > $1.timeStoppedRecording }).first,
+              environment
+              .workspacePath == nil ||
+              (referenceDate.timeIntervalSinceReferenceDate - 10 ..< referenceDate.timeIntervalSinceReferenceDate + 10) ~=
+              latestLog.timeStoppedRecording
+        else {
+            throw InspectBuildServiceError.noBuildLogFound(buildLogsPath: buildLogsPath, projectPath: projectPath)
+        }
+        let logPath = buildLogsPath.appending(component: latestLog.fileName)
+        return try xcactivityParser.parse(logPath)
+    }
+
+    private func projectPath(_ path: String?) async throws -> AbsolutePath {
+        if let workspacePath = environment.workspacePath {
+            if workspacePath.parentDirectory.extension == "xcodeproj" {
+                return workspacePath.parentDirectory
+            } else {
+                return workspacePath
+            }
+        } else {
+            let currentWorkingDirectory = try await fileSystem.currentWorkingDirectory()
```

Comment:
What's the current working directory when Xcode runs the scheme post-action?

Reply:
this code-path should never run in a post action as the scheme env variable should always be defined in that scenario.